base/iterators.jl

```diff
@@ -649,6 +649,58 @@ end
 iterate(it::Drop, state) = iterate(it.xs, state)
 isdone(it::Drop, state) = isdone(it.xs, state)
 
+
+# takewhile
+
+struct TakeWhile{I,P<:Function}
+    pred::P
+    xs::I
+end
+
+takewhile(pred,xs) = TakeWhile(pred,xs)
+
+function iterate(ibl::TakeWhile, itr...)
+    y = iterate(ibl.xs,itr...)
+    y === nothing && return nothing
+    ibl.pred(y[1]) || return nothing
+    y
+end
+
+IteratorSize(::Type{<:TakeWhile}) = SizeUnknown()
+eltype(::Type{TakeWhile{I,P}}) where {I,P} = eltype(I)
+IteratorEltype(::Type{TakeWhile{I,P}}) where {I,P} = IteratorEltype(I)
+
+
+# dropwhile
+
+struct DropWhile{I,P<:Function}
+    pred::P
+    xs::I
+end
+
+dropwhile(pred,itr) = DropWhile(pred,itr)
+
+function iterate(ibl::DropWhile,itr...)
+    if isempty(itr)
+        # one could use Iterators.filter with Iterators.Stateful
+        # but filter erases last elem state and
+        # stateful did not revert propertly due to nextvalstate/taken machinery
+        y = iterate(ibl.xs)
+        while y !== nothing
+            ibl.pred(y[1]) || break
+            y = iterate(ibl.xs,y[2])
+        end
+        y
+    else
+        iterate(ibl.xs, itr...)
+    end
+end
+
+IteratorSize(::Type{DropWhile{I,P}}) where {I,P} = SizeUnknown()
```

Comment:
Could write these and the others as

```suggestion
IteratorSize(::Type{<:DropWhile}) = SizeUnknown()
```

to not have to mention I and P everywhere.

Reply:
Jeff indicates that yesterday, i first fix it. But that raised a MethodError: no method matching length
May be because my code was not in base?
I suggest to let integration run and retry a fix.